funsor/testing.py

```diff
@@ -96,6 +96,10 @@ def assert_close(actual, expected, atol=1e-6, rtol=1e-6):
         and is_array(actual.terms[0].data)
     ):
         assert isinstance(expected, Contraction) and is_array(expected.terms[0].data)
+    elif isinstance(actual, Contraction) and isinstance(actual.terms[0], Delta):
+        assert isinstance(expected, Contraction) and isinstance(
+            expected.terms[0], Delta
+        )
```

Comment:
I'm not sure how to resolve this issue, where input `numpy.ndarray` is compared to `DeviceArray` but we can't assert they have the same type. This is not important I guess.

Reply:
Hmm, I think we should add some sort of check that values agree here, otherwise we'll have no confidence in the tests, right?

also nit: all assertions here should end with `, msg`

Reply:
Ah, I see the values are checked later in this function. LGTM.